New Bacterial genome annotation tutorial (#4677)
galaxyproject/training-material: topics/genome-annotation/tutorials/bacterial-genome-annotation/tutorial.md

time_estimation: 3H
key_points:
- Bakta is a powerful tool to annotate a bacterial genome
- Annotation can be easily visualized to understand the genomic context and help making sense of the annotations
tags:
- illumina
- bacteria
- microgalaxy
subtopic: prokaryote
level: Introductory

contributions:
  authorship:
  - bebatut
  funding:
  - abromics

follow_up_training:
- type: "internal"
  topic_name: genome-annotation
  tutorials:
  - apollo
- type: "internal"
  topic_name: visualisation
  tutorials:
  - circos
  - jbrowse
- type: "internal"
  topic_name: galaxy-interface
  tutorials:
  - history-to-workflow

---

After sequencing and assembly, a genome can be annotated. It is an essential step to describe the genome. 

Genome annotation consists in describing the structure and function of the components of the genome, by predicting, analyzing, and interpreting them in order to extract their biological significance and understand the biological processes in which they participate. Among other things, it identifies the locations of genes and all the coding regions in a genome (*structural annotation*) and determines what those genes do (*functional annotation*).

> To illustrate the process to identify AMR gene in a bacterial genome, we take an assembly of a bacterial genome (KUN1163 sample) generated by following a [bacterial genome assembly tutorial]({% link topics/assembly/tutorials/mrsa-illumina/tutorial.md %}) from data produced in "Complete Genome Sequences of Eight Methicillin-Resistant *Staphylococcus aureus* Strains Isolated from Patients in Japan" ({% cite Hikichi_2019 %}). 

Author: Sentence changed

> range from asymptomatic colonization of the nasal mucosa to soft tissue
> infection to fulminant invasive disease. Here, we report the complete
> genome sequences of eight MRSA strains isolated from patients in Japan.
{: .quote cite="{% cite_url Hikichi_2019 %}"}

> <agenda-title></agenda-title>
>
> In this tutorial, we will cover:
>
> 1. TOC
> {:toc}
>
{: .agenda}

# Galaxy and data preparation

Any analysis should get its own Galaxy history. So let's start by creating a new one and get the data (contig file from the assembly) into it.

> <hands-on-title>Prepare Galaxy and data</hands-on-title>
>
> 1. Create a new history for this analysis
>
>    {% snippet faqs/galaxy/histories_create_new.md %}
>
> 2. Rename the history
>
>    {% snippet faqs/galaxy/histories_rename.md %}
>
> 3. {% tool [Import](upload1) %} the contig file from [Zenodo]({{ page.zenodo_link }}) or from Galaxy shared data libraries:
>
>    ```
>    {{ page.zenodo_link }}/files/DRR187559_contigs.fasta
>    ```
>
>    {% snippet faqs/galaxy/datasets_import_via_link.md %}
>
>    {% snippet faqs/galaxy/datasets_import_from_data_library.md %}
>
>
{: .hands_on}

# Contig annotation

> For annotating the contigs, several tools exists to do that: Prokka ({% cite seemann2014prokka %}), Bakta ({% cite schwengers2021bakta %}), etc. Here, we use **Bakta** as recommended by Torsten Seeman as the successor of **Prokka**.

Reviewer: ```suggestion
For annotating the contigs, several tools exists to do that: Prokka ({% cite seemann2014prokka %}), Bakta ({% cite schwengers2021bakta %}), etc. Here, we use **Bakta** as recommended by {% include _includes/contributor-badge.html id="tseemann" %} as the successor of **Prokka**.
```

> **Bakta** is a tool for the rapid & standardized annotation of bacterial genomes and plasmids from both isolates and MAGs. It implements a comprehensive annotation workflow for coding and non-coding genes (*i.e.* tRNA, rRNA). 

Author: Added

> <hands-on-title>Contig annotation</hands-on-title>
>
> 1. {% tool [Bakta](toolshed.g2.bx.psu.edu/repos/iuc/bakta/bakta/1.8.2+galaxy0) %} with the following parameters:
>    - In *"Input/Output options"*:
>        - *"The bakta database"*: latest one
>        - *"The amrfinderplus database"*: latest one
>        - {% icon param-file %} *"Select genome in fasta format"*: Contig file
>    - In *"Optional annotation"*:
>        - *"Keep original contig header (--keep-contig-headers)"*: `Yes`
>    - In *"Selection of the output files"*:
>        - *"Output files selection"*: 
>          - `Annotation file in TSV`
>          - `Annotation and sequence in GFF3`
>          - `Feature nucleotide sequences as FASTA`
>          - `Summary as TXT`
>          - `Plot of the annotation result as SVG`
{: .hands_on}

**Bakta** can generate many outputs. Here we selected:

- `Analysis_summary`: a summary of the analysis as text file

   > <question-title></question-title>
   >
   > 1. How many contigs have been provided as input?
   > 2. How long is the draft genome?
   > 3. How many CDSs have been found?
   > 4. How many small proteins?
   > 5. Which other components have been found? 
   >

>    > How do it compare to results for KUN1163 in [Table 1](https://journals.asm.org/doi/10.1128/mra.01212-19#tab1) in {% cite Hikichi_2019 %}?

Author: Added few more words about that

>    > How do it compare to results for KUN1163 in [Table 1](https://journals.asm.org/doi/10.1128/mra.01212-19#tab1) in {% cite Hikichi_2019 %}?

Reviewer 2: hm, is it commited? I can't see it

also:
```suggestion
   > How does it compare to results for KUN1163 in [Table 1](https://journals.asm.org/doi/10.1128/mra.01212-19#tab1) in {% cite Hikichi_2019 %}?
```

>    > How do it compare to results for KUN1163 in [Table 1](https://journals.asm.org/doi/10.1128/mra.01212-19#tab1) in {% cite Hikichi_2019 %}?

Author: The additions were directly in the answers to the specific questions (1, 2, 3, etc)

   2. attC recombination sites with a covariance model (CM), which models the secondary structure in addition to the few conserved sequence positions.

3. Integrates the results to distinguish 3 types of elements

   - Complete integron: Integron with integron integrase nearby attC site(s)
   - In0 element: Integron integrase only, without any attC site nearby
   - CALIN element: Cluster of attC sites Lacking INtegrase nearby 

> <hands-on-title> Integron identification</hands-on-title>
>
> 1. {% tool [IntegronFinder](toolshed.g2.bx.psu.edu/repos/iuc/integron_finder/integron_finder/2.0.2+galaxy1) %} with the following parameters:
>    - {% icon param-file %} *"Replicon file"*: Contig file
>    - *"Thorough local detection"*: `Yes`
>    - *"Search also for promoter and attI sites?"*: `Yes`
>    - *"Remove log file"*: `Yes`
>
{: .hands_on}

**IntegronFinder** generates 2 outputs:

- A summary with for each sequence in the input the number of identified CALIN elements, In0 elements, and complete integrons.

   > <question-title></question-title>
   >
   > How many integron elements have been found?
   >
   > > <solution-title></solution-title>
   > >

>    > > No integron elements have been found on any contig.

Reviewer 2: Why? :D

>                    - *"Track Type"*: `GFF/GFF3/BED Features`
>                        - {% icon param-file %} *"GFF/GFF3/BED Track Data"*: GFF output of **ISEScan**
>                        - *"JBrowse Track Type [Advanced]"*: `Neat Canvas Features`
>                        - *"Track Visibility"*: `On for new users`
>
>        *If integrons are found as **IntegronFinder***
>
>        - {% icon param-repeat %} *"Insert Track Group"*
>            - *"Track Category"*: `Integrons`
>            - In *"Annotation Track"*:
>                - {% icon param-repeat %} *"Insert Annotation Track"*
>                    - *"Track Type"*: `GFF/GFF3/BED Features`
>                        - {% icon param-file %} *"GFF/GFF3/BED Track Data"*: `IntegronFinder GFF`
>                        - *"JBrowse Track Type [Advanced]"*: `Neat Canvas Features`
>                        - *"Track Visibility"*: `On for new users`
>
> 3. View the output of JBrowse
>
{: .hands_on}

In the output of the JBrowse you can view the genes, IS, plasmid, etc on the contigs. With the search tools you can easily find genes of interest. JBrowse can handle many inputs and can be very useful.

> If it takes too long to build the JBrowse instance, you can view an embedded one here. (**Warning**: feature name search will not work.)

Reviewer 2: Need a link here?

Author: It is embedded directly there. Do we still need a link?

Reviewer 2: no link is fine indeed

> {% snippet topics/visualisation/faqs/visualizations_jbrowse.html datadir="data" loc="contig00019:1..30347" tracks="DNA,7e37ab7a5f0a1610de9d213b6a714666_0,dd555b269a8894688c3c7551d0d6a868_0,e1c36f2c4382e60836c6c4b4d8026fa5_0" %}

Reviewer 2: Might need to say that it's ok to have the `Error reading from name store.` error

Author: Added a comment box with that

   2. attC recombination sites with a covariance model (CM), which models the secondary structure in addition to the few conserved sequence positions.

3. Integrates the results to distinguish 3 types of elements

   - Complete integron: Integron with integron integrase nearby attC site(s)
   - In0 element: Integron integrase only, without any attC site nearby
   - CALIN element: Cluster of attC sites Lacking INtegrase nearby 

> <hands-on-title> Integron identification</hands-on-title>
>
> 1. {% tool [IntegronFinder](toolshed.g2.bx.psu.edu/repos/iuc/integron_finder/integron_finder/2.0.2+galaxy1) %} with the following parameters:
>    - {% icon param-file %} *"Replicon file"*: Contig file
>    - *"Thorough local detection"*: `Yes`
>    - *"Search also for promoter and attI sites?"*: `Yes`
>    - *"Remove log file"*: `Yes`
>
{: .hands_on}

**IntegronFinder** generates 2 outputs:

- A summary with for each sequence in the input the number of identified CALIN elements, In0 elements, and complete integrons.

   > <question-title></question-title>
   >
   > How many integron elements have been found?
   >
   > > <solution-title></solution-title>
   > >

>    > > No integron elements have been found on any contig. It could be because the genome is too stable, or because the assembly is not good enough quality and some parts useful for the integron detection were removed.

Reviewer 2: ```suggestion
   > > No integron elements have been found on any contig. It could be because the genome is too stable, or because the assembly quality is not good enough and some parts useful for the integron detection were removed.
```
